src/gui/editors/PianoRoll.cpp

```diff
@@ -4165,6 +4165,13 @@ void PianoRoll::pasteNotes()
 			cur_note.restoreState( list.item( i ).toElement() );
 			cur_note.setPos( cur_note.pos() + Note::quantized( m_timeLine->pos(), quantization() ) );
 
+			// handle case where notes are copied from BB editor
+			if (cur_note.length().getTicks() == -192)
+			{
+				// magic for addNote to indicate that paste came from BB editor
+				cur_note.setLength(MidiTime( -193 ));
```

Comment:
We should make the `-193` a magic number constant, so something like `BB_NOTE_COPYPASTE_LENGTH`.

Reply:
I've added the magic number in Pattern.h